lib/widgets/cards/report_card.dart

```diff
@@ -175,7 +176,8 @@ class _ReportCardState extends State<ReportCard> {
 							materialTapTargetSize: MaterialTapTargetSize.shrinkWrap,
 							padding: EdgeInsets.symmetric(vertical: 8.0),
 							colorBrightness: Brightness.dark,
-							onPressed: () { 
+							onPressed: () async {
+								widget.returnCallback();
```

Comment:
That's being called when entering the report form, not when returning from it - is that deliberate? If so you should change the naming to be less confusing :wink: